src/nix/export.cc

```diff
@@ -0,0 +1,126 @@
+#include "command.hh"
+#include "common-args.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include "fs-accessor.hh"
+
+using namespace nix;
+
+struct CmdExport : InstallableCommand
+{
+    std::string exporter = "github:matthewbauer/nix-bundle";
+    Path outLink;
+
+    CmdExport()
+    {
+        addFlag({
+            .longName = "exporter",
+            .description = "use custom exporter",
+            .labels = {"flake-url"},
+            .handler = {&exporter},
+            .completer = {[&](size_t, std::string_view prefix) {
+                completeFlakeRef(getStore(), prefix);
+            }}
+        });
+
+        addFlag({
+            .longName = "out-link",
+            .shortName = 'o',
+            .description = "path of the symlink to the build result",
+            .labels = {"path"},
+            .handler = {&outLink},
+            .completer = completePath
+        });
+    }
+
+    std::string description() override
+    {
+        return "export an application out of the Nix store";
+    }
+
+    Examples examples() override
+    {
+        return {
+            Example{
+                "To export Hello:",
+                "nix export hello"
+            },
+        };
+    }
+
+    Strings getDefaultFlakeAttrPaths() override
+    {
+        Strings res{"defaultApp." + settings.thisSystem.get()};
+        for (auto & s : SourceExprCommand::getDefaultFlakeAttrPaths())
+            res.push_back(s);
+        return res;
+    }
+
+    Strings getDefaultFlakeAttrPathPrefixes() override
+    {
+        Strings res{"apps." + settings.thisSystem.get() + ".", "packages"};
+        for (auto & s : SourceExprCommand::getDefaultFlakeAttrPathPrefixes())
+            res.push_back(s);
+        return res;
+    }
+
+    void run(ref<Store> store) override
+    {
+        auto evalState = getEvalState();
+
+        auto app = installable->toApp(*evalState);
+        store->buildPaths(app.context);
+
+        auto [exporterFlakeRef, exporterName] = parseFlakeRefWithFragment(exporter, absPath("."));
+        const flake::LockFlags lockFlags{ .writeLockFile = false };
+        auto exporter = InstallableFlake(
+            evalState, std::move(exporterFlakeRef),
+            Strings{exporterName == "" ? ("defaultExporter." + settings.thisSystem.get()) : exporterName},
+            Strings({"exporters." + settings.thisSystem.get() + "."}), lockFlags);
```

Comment:
Given that this is a function anyway, we don't need to use the `.${system}` attribute. We can just pass `system` as a function argument. So the interface of the exporter would be `{ system, program, ... }: ...`.